nomenclature/config.py

```diff
@@ -1,18 +1,69 @@
 from pathlib import Path
-from typing import Dict, Optional
-from pydantic import BaseModel
+from typing import Optional, Dict
+from pydantic import BaseModel, root_validator, validator
 
 import yaml
+from git import Repo
 
 
 class CodeListConfig(BaseModel):
-    repository: Optional[Path]
+    dimension: str
+    repository: Optional[str]
+    repository_definition_path: Optional[str]
```

Comment:
This does not have the "definitions" path, but the dimension within a definitions folder...

Also, should this be a `Path` object?
```suggestion
    repository_dimension_path: Optional[Path]
```